File: server/server/models/Provider.js
const mongoose = require("mongoose");

const providerSchema = new mongoose.Schema({
  name: { type: String, required: true },
  phone: { type: String, required: true, unique: true },
  skills: [String],                        // ["Plumber", "Electrician"]
  category: String,                        // "Repairs"
  hourlyRate: Number,
  bio: String,
  photo: String,
  languages: [String],                     // ["Kannada", "Hindi"]
  rating: { type: Number, default: 0 },
  totalReviews: { type: Number, default: 0 },
  isVerified: { type: Boolean, default: false },
  isAvailable: { type: Boolean, default: true },

  // 🗺️ GeoJSON for 5km proximity search
  location: {
    type: { type: String, enum: ["Point"], default: "Point" },
    coordinates: [Number],               // [longitude, latitude]
    address: String,
  },
}, { timestamps: true });

// Required for $nearSphere queries
providerSchema.index({ location: "2dsphere" });

module.exports = mongoose.model("Provider", providerSchema);

File: server/app/src/main/java/com/example/gigmarket/data/model/Provider.kt
package com.example.gigmarket.data.model

data class Provider(
    val _id: String,
    val name: String,
    val phone: String,
    val skills: List<String>,
    val category: String,
    val hourlyRate: Double,
    val bio: String?,
    val photo: String?,
    val languages: List<String>,
    val rating: Double,
    val totalReviews: Int,
    val isVerified: Boolean,
    val isAvailable: Boolean,
    val location: LocationModel
)

data class LocationModel(
    val type: String,
    val coordinates: List<Double>, // [longitude, latitude]
    val address: String?
)

data class NearbyProvidersResponse(
    val success: Boolean,
    val count: Int,
    val providers: List<Provider>
)

data class ProviderResponse(
    val success: Boolean,
    val provider: Provider
)
